### ISSUE TITLE
[BUG] Snowcat does not support labels

### ISSUE THREAD

**Issue by lemmy** [informalsystems/apalache#653]
```shell
markus@avocado:~/src/TLA/_specs/models/tutorials/ewd840(main)$ apalache --inv=Inv --length=15 --with-snowcat check EWD840
# Tool home: /home/markus/src/TLA/_community/apalache
# Package:   /home/markus/src/TLA/_community/apalache/mod-distribution/target/apalache-pkg-0.11.1-SNAPSHOT-full.jar
# JVM args: -Xmx4096m -DTLA-Library=/home/markus/src/TLA/_community/apalache/src/tla
#
# APALACHE version 0.11.1-SNAPSHOT build v0.11.0-22-g5237650
#
# WARNING: This tool is in the experimental stage.
#          Please report bugs at: [https://github.com/informalsystems/apalache/issues]
# 
# Usage statistics is ON. Thank you!
# If you have changed your mind, disable the statistics with config --enable-stats=false.

Checker options: filename=EWD840, init=, next=, inv=Inv           I@13:38:38.690
Tuning:                                                           I@13:38:39.134
PASS #0: SanyParser                                               I@13:38:39.135
Parsing file /home/markus/src/TLA/_specs/models/tutorials/ewd840/EWD840.tla
Parsing file /tmp/Naturals.tla
PASS #1: ConfigurationPass                                        I@13:38:39.777
  > EWD840: Loading TLC configuration                             I@13:38:39.779
  > No TLC configuration found. Skipping.                         I@13:38:39.786
  > Command line option --init is not set. Using Init             I@13:38:39.787
  > Command line option --next is not set. Using Next             I@13:38:39.787
  > Set the initialization predicate to Init                      I@13:38:39.787
  > Set the transition predicate to Next                          I@13:38:39.788
  > Set an invariant to Inv                                       I@13:38:39.789
PASS #2: TypeCheckerSnowcat                                       I@13:38:39.833
 > Running Snowcat .::.                                           I@13:38:39.833
(Please report an issue at: [https://github.com/informalsystems/apalache/issues],java.lang.IllegalArgumentException: Unsupported expression: "P0":: ∀i ∈ Node(): ((tpos < i) ⇒ ¬active[i]))
Unhandled exception                                               E@13:38:39.901
java.lang.IllegalArgumentException: Unsupported expression: "P0":: ∀i ∈ Node(): ((tpos < i) ⇒ ¬active[i])
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.transform(ToEtcExpr.scala:910)
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.apply(ToEtcExpr.scala:153)
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.$anonfun$transform$1(ToEtcExpr.scala:174)
	at scala.collection.immutable.List.map(List.scala:293)
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.mkExRefApp$1(ToEtcExpr.scala:174)
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.transform(ToEtcExpr.scala:254)
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.apply(ToEtcExpr.scala:153)
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.operDefToDecl(ToEtcExpr.scala:78)
	at at.forsyte.apalache.tla.typecheck.etc.ToEtcExpr.apply(ToEtcExpr.scala:44)
	at at.forsyte.apalache.tla.typecheck.TypeCheckerTool.$anonfun$check$1(TypeCheckerTool.scala:33)
	at scala.collection.immutable.List.$anonfun$foldRight$1(List.scala:447)
	at scala.collection.immutable.List.foldRight(List.scala:91)
	at at.forsyte.apalache.tla.typecheck.TypeCheckerTool.check(TypeCheckerTool.scala:32)
	at at.forsyte.apalache.tla.typecheck.TypeCheckerTool.checkAndTag(TypeCheckerTool.scala:62)
	at at.forsyte.apalache.tla.typecheck.passes.EtcTypeCheckerPassImpl.execute(EtcTypeCheckerPassImpl.scala:65)
	at at.forsyte.apalache.infra.passes.PassChainExecutor.exec$1(PassChainExecutor.scala:22)
	at at.forsyte.apalache.infra.passes.PassChainExecutor.run(PassChainExecutor.scala:37)
	at at.forsyte.apalache.tla.Tool$.runCheck(Tool.scala:177)
	at at.forsyte.apalache.tla.Tool$.$anonfun$run$2(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.$anonfun$run$2$adapted(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.handleExceptions(Tool.scala:264)
	at at.forsyte.apalache.tla.Tool$.run(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.main(Tool.scala:45)
	at at.forsyte.apalache.tla.Tool.main(Tool.scala)
It took me 0 days  0 hours  0 min  1 sec                          I@13:38:39.902
Total time: 1.215 sec                                             I@13:38:39.903
EXITCODE: ERROR (99)
```

```tla
------------------------------- MODULE EWD840 -------------------------------
EXTENDS Naturals

CONSTANT N
ASSUME N \in (Nat \ {0})

Node == 0 .. N-1

black == "black"
white == "white"
Color == {black, white}

VARIABLES active, color, tpos, tcolor
vars == <<active, color, tpos, tcolor>>

TypeOK ==
  /\ active \in [Node -> BOOLEAN]
  /\ color \in [Node -> Color]
  /\ tpos \in Node
  /\ tcolor \in Color

Init ==
  /\ active \in [Node -> BOOLEAN]
  /\ color \in [Node -> Color]
  /\ tpos \in Node
  /\ tcolor = black

InitiateProbe ==
  /\ tpos = 0
  /\ \/ tcolor = black
     \/ color[0] = black
  /\ tpos' = N-1
  /\ tcolor' = white
  /\ color' = [color EXCEPT ![0] = white]
  /\ UNCHANGED <<active>>

PassToken(i) ==
    /\  \/ ~active[i]
        \* There is no need to wait for this node to terminate if a
        \* higher-numbered node has already marked this round inconclusive.
        \* Instead, the initiator may start a new round (upon receipt of
        \* the token) in the meantime.
        \/ tcolor = black
        \* Likewise, don't wait for this node to terminate, if it is
        \* going to cause an inconclusive round.
        \/ color[i] = black
    /\ tpos = i
    /\ tpos' = i-1
    \* Passing along the token transfers the taint
    \* from the node to the token.
    /\ tcolor' = IF color[i] = black THEN black ELSE tcolor
    /\ color' = [color EXCEPT ![i] = white]
    /\ UNCHANGED <<active>>

System == InitiateProbe \/ \E i \in Node \ {0} : PassToken(i)

-----------------------------------------------------------------------------

Deactivate(i) ==
    /\ active[i]
    /\ active' = [active EXCEPT ![i] = FALSE]
    /\ UNCHANGED <<color, tpos, tcolor>>

SendMsg(i) ==
    /\ active[i]
    /\ \E j \in Node:
            /\ active' = [active EXCEPT ![j] = TRUE]
            \* If the message activates a lower-numbered node, there is
            \* no need to taint the token to mark the round inconclusive.
            \* This is because the lower numbered node has not yet received
            \* the token for this round and, thus, can taint it itself.
            \* Thus, this is an optimization to avoid another round in the
            \* case that the recipient node deactivates between receiving
            \* the message and the token.
            /\ color' = [color EXCEPT ![i] = IF j>i THEN black ELSE @]
    /\ UNCHANGED <<tpos, tcolor>>

Environment ==
    \E i \in Node:
        \/ Deactivate(i)
        \/ SendMsg(i)

Next == System \/ Environment

Spec == Init /\ [][Next]_vars /\ WF_vars(System)

-----------------------------------------------------------------------------

terminationDetected ==
    /\ ~active[0]
    /\ color[0] = white
    /\ tpos = 0
    /\ tcolor = white

terminated ==
    \A n \in Node: ~active[n]

\* If termination has been detected, all nodes are
\* indeed inactive.
Inv == 
    terminationDetected => terminated

THEOREM Spec => []Inv

-----------------------------------------------------------------------------

\* Dijkstra's (inductive) invariant.
IInv ==
    \/ P0:: \A i \in Node : tpos < i => ~ active[i]
    \/ P1:: \E j \in 0 .. tpos : color[j] = black
    \/ P2:: tcolor = black

THEOREM Spec => []IInv

-----------------------------------------------------------------------------

\* Termination of nodes leads to termination being detected.
Prop ==
    terminated ~> terminationDetected

THEOREM Spec => Prop

-----------------------------------------------------------------------------

SpecWFEnv ==
    Spec /\ WF_vars(Environment)

AlwaysTerminates ==    
    <>[](terminationDetected)

THEOREM SpecWFEnv => AlwaysTerminates

=============================================================================
```

**Comment by konnov** [comment]
Yep. I have also found that when integrating types. It is implemented in #657 but you will have to wait a bit.

**Comment by konnov** [comment]
Now it is in unstable.
